Tests/Common/Securities/SecurityTransactionManagerTests.cs

```diff
@@ -0,0 +1,124 @@
+
+            algorithm.Securities = securities;
+            transactionHandler.Initialize(algorithm, new BacktestingBrokerage(algorithm), _resultHandler);
+            transactions.SetOrderProcessor(transactionHandler);
+
+            // Adding cash: strike price times number of shares
+            portfolio.SetCash(192 * 100);
+
+            securities.Add(
+                Symbols.SPY,
+                new Security(
+                    SecurityExchangeHours,
+                    CreateTradeBarDataConfig(Symbols.SPY),
+                    new Cash(Currencies.USD, 0, 1m),
+                    SymbolProperties.GetDefault(Currencies.USD),
+                    ErrorCurrencyConverter.Instance,
+                    RegisteredSecurityDataTypesProvider.Null,
+                    new SecurityCache()
+                )
+            );
+            securities.Add(
+                Symbols.IBM,
+                new Security(
+                    SecurityExchangeHours,
+                    CreateTradeBarDataConfig(Symbols.IBM),
+                    new Cash(Currencies.USD, 0, 1m),
+                    SymbolProperties.GetDefault(Currencies.USD),
+                    ErrorCurrencyConverter.Instance,
+                    RegisteredSecurityDataTypesProvider.Null,
+                    new SecurityCache()
+                )
+            );
+            securities[Symbols.IBM].Holdings.SetHoldings(1, 1);
+            securities[Symbols.SPY].Holdings.SetHoldings(1, 1);
+
+            var holdings = securities[Symbols.IBM].Holdings.Quantity;
+            transactions.AddOrder(new SubmitOrderRequest(OrderType.Market, SecurityType.Equity, Symbols.IBM, -holdings, 0, 0, securities.UtcTime, ""));
+            transactions.AddOrder(new SubmitOrderRequest(OrderType.Market, SecurityType.Equity, Symbols.SPY, -holdings, 0, 0, securities.UtcTime, ""));
+
+            Func<Order, bool> basicOrderFilter = x => true;
+            Func<OrderTicket, bool> basicOrderTicketFilter = x => true;
+            using (Py.GIL())
+            {
+                var orders = transactions.GetOrders(basicOrderFilter.ToPython());
+                var orderTickets = transactions.GetOrderTickets(basicOrderTicketFilter.ToPython());
+                var openOrders = transactions.GetOpenOrders(basicOrderFilter.ToPython());
+                var openOrdersTickets = transactions.GetOpenOrderTickets(basicOrderTicketFilter.ToPython());
+                var openOrdersRemaining = transactions.GetOpenOrdersRemainingQuantity(basicOrderTicketFilter.ToPython());
+                Assert.AreEqual(2, orders.Count());
+                Assert.AreEqual(2, orderTickets.Count());
+                Assert.AreEqual(0, openOrders.Count);
+                Assert.AreEqual(0, openOrdersTickets.Count());
+                Assert.AreEqual(0, openOrdersRemaining);
+                foreach(var ticket in orderTickets)
+                {
+                    Log.Trace($"Ticket symbol: {ticket.Symbol} - Status: {ticket.Status}");
```

Comment:
Same comment about logging 👍 generally tests should log rarely in debug mode only. 
I think the unit test could be simplified, really needs a `SecurityTransactionManager` and some `TestOrderProcessor`, can create it passing in the orders you want I think